youtube_dl/extractor/heise.py

```diff
@@ -88,3 +107,49 @@ def _real_extract(self, url):
                 self._html_search_meta('date', webpage)),
             'formats': formats,
         }
+
+    def ctUplinkHelper(self, title, video_id):
+        # e.g. "18.5" from "c't uplink 18.5:"
+        episode_str = re.findall(r'[0-9]{1,2}.[0-9]{1,2}', title)
+
+        feeds = [
+            self._download_xml(
+                'https://www.heise.de/ct/uplink/ctuplink.rss',
+                video_id, "Downloading alternative XML (audio)"),
+            self._download_xml(
+                'https://www.heise.de/ct/uplink/ctuplinkvideo.rss',
+                video_id, "Downloading alternative XML (SD)"),
+            self._download_xml(
+                'https://www.heise.de/ct/uplink/ctuplinkvideohd.rss',
+                video_id, "Downloading alternative XML (HD)")]
```

Comment:
1. Code duplication.
2. Each should be non fatal.

Reply:
regarding 1.: `_download_xml` only takes one URL, doesn't it? To be honest I don't know how to shrink this further, mainly because I don't know what specificly you are referring to. Even though I see there is some duplication. Can you give me a hint?

Reply:
```python
for path, format_id in (('', 'audio'), ('video', 'sd'), ('videohd', 'hd')):
    self._download_xml(
        'https://www.heise.de/ct/uplink/ctuplink%s.rss' % path,
        video_id, 'Downloading %s XML' % format_id, fatal=False)
```